docs/proposals/20210628-nodepool-ingress-support.md

```diff
@@ -186,45 +239,149 @@ As an end user, I want to balance the traffic into a NodePool with layer-7 load-
-	- A new CRD will be defined to provide the daemonset support from NodePool level, so that one ingress controller
-	  can be deployed to the NodePool which needs to enable ingress feature.
-	- For every ingress controller deployment in different NodePools, it creates a NodePort type service to expose the
-	  ingress controller service, and binds a corresponding configmap to differentiate the NodePools. All the ingress
-	  controller deployments for different NodePools can use one same service account.
-	- Add 2 NodePool annotations for users to configure the ingress feature when NodePool is created:
-	  ```yaml
-            nodepool.openyurt.io/ingress-ips: "xxx"      # the ingress controller service IP scope
-            nodepool.openyurt.io/ingress-config: "xxx"   # the ingress controller nginx config
-	  ```
-
-2). Nginx ingress controller:
-	- Keep the upstream version none-intrusive, so that it will be very convenient to upgrade in future.
-	- It specifies the ingress class to NodePool name so the ingress CR can select the corresponding ingress controller.
-	- It should be granted the authorities to create/update/patch the related resources in kube-system namespace.
-	- It connects with Yurthub to monitor the resources(ingress/service/configmap/secret/endpoint) of its own NodePool.
-
-3). Ingress controller sidecar:
-	- When sidecar is launched, it gets the NodePool id it belongs to, and connects to Yurthub to list/watch the NodePool.
-	- When it detects the NodePool annotation below is changed, it will update externalIPs of the corresponding service:
-	  ```yaml
-            nodepool.openyurt.io/ingress-ips: "xxx"
-	  ```
-	- When it detects the NodePool annotation below is changed, it will update the corresponding configmap of the NodePool,
-	  which will trigger the ingress controller to reload nginx config for the resource state sync.
-	  ```yaml
-            nodepool.openyurt.io/ingress-config: "xxx"
-	  ```
+#### 1). Yurt-app-manager
+
+Implement a standalone `YurtIngress` operator to support the NodePool ingress feature, which means that `YurtIngress`
+together with `NodePool/YurtAppSet(UnitedDeployment)/YurtAppDaemon` will compose the the current OpenYurt unitization
+capability, and they are all managed by Yurt-app-manager. The responsibility of `YurtIngress` is to monitor the update
+of `YurtIngress` CR and trigger the corresponding operations to nginx ingress controller related components.
+
+`YurtIngress` CRD definition:
+
+```go
+type IngressNotReadyType string
+
+const (
+	IngressPending IngressNotReadyType = "Pending"
+	IngressFailure IngressNotReadyType = "Failure"
+)
+
+// IngressPool defines the details of a Pool for ingress
+type IngressPool struct {
+	// Indicates the pool name.
+	Name string `json:"name"`
+
+	// Pool specific configuration will be supported in future
+}
+
+// IngressNotReadyPool defines the condition details of an ingress not ready Pool
+type IngressNotReadyPool struct {
```

Comment:
`IngressNotReadyPool` is unused?

Reply:
It is used in YurtIngressCondition, I missed the update in it, I will update it soon.